### ISSUE TITLE
Broken cache invalidation in defaultPipelinePrepare

### ISSUE THREAD

**Issue by bkupidura** [axa-group/nlp.js#543]
**Describe the bug**
Current cache invalidation in defaultPipelinePrepare is broken.
```
UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'en' of undefined
    at NeuralNlu.defaultPipelinePrepare (/data/node_modules/@nlpjs/nlu/src/nlu.js:96:34)
```

**To Reproduce**
Im using node-red chatbot, so no pure JS code to reproduce.

**Additional context**
Bug in chatbot https://github.com/guidone/node-red-contrib-chatbot/issues/434

Currently when cache is older than 1h, nlp.js is setting `this.cache.results` to undefined. Afaik it should create empty object (or set whole `this.cache` to undefined, which will lead to recreate cache object from scratch).

Please check https://github.com/axa-group/nlp.js/blob/master/packages/nlu/src/nlu.js#L82 and https://github.com/axa-group/nlp.js/blob/master/packages/nlu/src/nlu.js#L96

**Comment by jesus-seijas-sp** [comment]
Hello! First at all thank you, I'm preparing the hotfix, as soon as Travis ends confirmation I'll publish a new version with this fix.

**Comment by jesus-seijas-sp** [comment]
Published hotfix as 4.10.4
Packages affected:
 - @nlpjs/basic@4.10.4
 - @nlpjs/nlp@4.10.4
 - @nlpjs/nlu-luis@4.10.4
 - @nlpjs/nlu@4.10.4
 - node-nlp@4.10.4

**Comment by jesus-seijas-sp** [comment]
closing: solved in 4.10.4
